### DISCUSSION TITLE
Kha newbie here :) Some initial questions...

### DISCUSSION THREAD

**Discussion by tjrhodes** [Kode/Kha#1325] [Q&A]
Hi,

I'm a long time haxer but have never dabbled with Kha before today. I've just started with the graphics stuff and would like to know the following:

1. Is it possibile to smooth the output drawn to the framebuffer? Thin outlined circles for example look really jagged.
2. Is it possibile to grab the bytes of a buffer to write out to create a series of pngs?

Thanks, looks awesome btw so far. 

Cheers, Tom.

**Comment by RobDangerous** [comment]
1. Kha is a hardware-abstraction/portability system - there is no 2D vector graphics hardware to speak of out there so that's not part of Kha aside from some very simple things so we can make something that can run reasonably well on html-canvas as well as modern graphics APIs. GraphicsExtension.hx is misleading in that regard - sorry about that, it will be removed once https://github.com/RobDangerous/nanovg is working well enough.
There's a setting for multi-sampling in G4 but that doesn't work everywhere and you should better understand what it is before using it.
2. Image.getPixels works for render-targets. A PNG encoder is not included.

**Reply by tjrhodes** [reply]
Ok, Thanks Rob.

So no smooth vectors then until nanovg? That looks cool btw.

Right, so I could render everything to an Image with createRenderTarget and then get the pixels and use a lib to create a png. 

Thanks for the quick reply, for a poss upcoming project I need smooth vectors but will definitely be getting more in to Kha, looks awesome, looks to deliver on the cross target potential of a haxe codebase. 

Cheers, Tom.
